spec/index.bs

```diff
@@ -10184,53 +10270,68 @@ GPURenderPassEncoder includes GPUBindingCommandsMixin;
 
-    : <dfn>\[[attachment_size]]</dfn>
+    : <dfn>\[[attachment_size]]</dfn>, readonly
     ::
         Set to the following extents:
 
         - `width, height` = the dimensions of the pass's render attachments
 
-    : <dfn>\[[occlusion_query_set]]</dfn>, of type {{GPUQuerySet}}.
+    : <dfn>\[[occlusion_query_set]]</dfn>, of type {{GPUQuerySet}}, readonly
     ::
         The {{GPUQuerySet}} to store occlusion query results for the pass, which is initialized with
         {{GPURenderPassDescriptor}}.{{GPURenderPassDescriptor/occlusionQuerySet}} at pass creation time.
 
-    : <dfn>\[[occlusion_query_active]]</dfn>, of type {{boolean}}.
+    : <dfn>\[[occlusion_query_active]]</dfn>, of type {{boolean}}
     ::
         Whether the pass's {{GPURenderPassEncoder/[[occlusion_query_set]]}} is being written.
 
+    : <dfn>\[[endTimestampWrites]]</dfn>, of type {{GPURenderPassTimestampWrites}}, readonly
+    ::
+        The timestamp attachments which need to be executed when the pass ends.
+
+    : <dfn>\[[maxDrawCount]]</dfn> of type {{GPUSize64}}, readonly
+    ::
+        The maximum number of draws allowed in this pass.
+</dl>
+
+When executing encoded render pass commands as part of a {{GPUCommandBuffer}}, an internal
+<dfn dfn>RenderState</dfn> object is used to track the current state required for rendering.
+
+[=RenderState=] contains the following internal slots used for execution of rendering commands:
+
+<dl dfn-type=attribute dfn-for=RenderState>
+    : <dfn>\[[occlusionQueryIndex]]</dfn>, of type {{GPUSize32}}.
+    ::
+        The index into {{GPURenderPassEncoder/[[occlusion_query_set]]}} at which to store the
+        occlusion query results.
+
     : <dfn>\[[viewport]]</dfn>
     ::  Current viewport rectangle and depth range.
 
-    : <dfn>\[[blend_constant]]</dfn>
+    : <dfn>\[[scissorRect]]</dfn>
+    ::  Current scissor rectangle.
+
+    : <dfn>\[[blendConstant]]</dfn>
     ::  Current blend constant value, initially `[0, 0, 0, 0]`.
 
-    : <dfn>\[[stencil_reference]]</dfn>
+    : <dfn>\[[stencilReference]]</dfn>
     ::  Current stencil reference value, initially `0`.
-
-    : <dfn>\[[endTimestampWrites]]</dfn>, of type {{GPURenderPassTimestampWrites}}
-    ::
-        The timestamp attachments which need to be executed when the pass ends.
-
-    : <dfn>\[[maxDrawCount]]</dfn> of type {{GPUSize64}}.
-    ::
-        The maximum number of draws allowed in this pass.
 </dl>
 
-When a {{GPURenderPassEncoder}} is created, it has the following default state:
+When a [=RenderState=] is created, it has the following default state:
 
-- {{GPURenderPassEncoder/[[viewport]]}}:
+- {{RenderState/[[viewport]]}}:
     - `x, y` = `0.0, 0.0`
     - `width, height` = the dimensions of the pass's render targets
     - `minDepth, maxDepth` = `0.0, 1.0`
-- Scissor rectangle:
+- {{RenderState/[[scissorRect]]}}:
     - `x, y` = `0, 0`
     - `width, height` = the dimensions of the pass's render targets
```

Comment:
Aside: Pretty sure this bullet list predates the `<dl>` block, let's move these defaults into the definitions